src/main/java/com/myhome/services/springdatajpa/HouseMemberDocumentSDJpaService.java

```diff
@@ -0,0 +1,144 @@
+
+import javax.imageio.IIOImage;
+import javax.imageio.ImageIO;
+import javax.imageio.ImageWriteParam;
+import javax.imageio.ImageWriter;
+import javax.imageio.stream.ImageOutputStream;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.nio.file.Files;
+import java.util.Optional;
+
+@Service
+public class HouseMemberDocumentSDJpaService implements HouseMemberDocumentService {
+
+    @Value("${server.files.compressionBorderSizeKBytes}")
+    private int compressionBorderSizeKBytes;
+
+    @Value("${server.files.maxSizeKBytes}")
+    private int maxFileSizeKBytes;
+
+    @Value("${server.files.compressedImageQuality}")
+    private int compressedImageQuality;
+
+    private final HouseMemberRepository houseMemberRepository;
+    private final HouseMemberDocumentRepository houseMemberDocumentRepository;
+
+    public HouseMemberDocumentSDJpaService(HouseMemberRepository houseMemberRepository, HouseMemberDocumentRepository houseMemberDocumentRepository) {
+        this.houseMemberRepository = houseMemberRepository;
+        this.houseMemberDocumentRepository = houseMemberDocumentRepository;
+    }
+
+    @Override
+    public Optional<HouseMemberDocument> findHouseMemberDocument(String memberId) {
+        HouseMember member = houseMemberRepository.findByMemberId(memberId);
+        if(member != null) {
+            return Optional.ofNullable(member.getHouseMemberDocument());
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    public boolean deleteHouseMemberDocument(String memberId) {
+        boolean documentDeleted = false;
+        HouseMember member = houseMemberRepository.findByMemberId(memberId);
+        if (member != null) {
+            houseMemberDocumentRepository.delete(member.getHouseMemberDocument());
+            member.setHouseMemberDocument(null);
+            houseMemberRepository.save(member);
+            documentDeleted = true;
+        }
+        return documentDeleted;
+    }
+
+    @Override
+    public boolean updateHouseMemberDocument(MultipartFile multipartFile, String memberId) throws IOException {
+        return createHouseMemberDocument(multipartFile, memberId);
+    }
+
+    @Override
+    public boolean createHouseMemberDocument(MultipartFile multipartFile, String memberId) throws IOException {
+        boolean documentCreated = false;
+        HouseMember member = houseMemberRepository.findByMemberId(memberId);
+        if (member != null) {
+            documentCreated = tryCreateDocument(multipartFile, member);
+        }
+        return documentCreated;
+    }
+
+    private boolean tryCreateDocument(MultipartFile multipartFile, HouseMember member) throws IOException {
+
+        boolean documentCreated = false;
+        File memberDocumentFile = createMemberDocumentFile(member);
+
+        BufferedImage documentImage = getImageFromMultipartFile(multipartFile);
+        if (multipartFile.getSize() < 1024 * compressionBorderSizeKBytes) {
```

Comment:
Magic number: 1024

https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad